api/src/main/java/io/cloudevents/impl/CloudEventImpl.java

```diff
@@ -38,22 +39,60 @@ public CloudEventImpl(Attributes attributes, byte[] data, Map<String, Object> ex
         this.extensions = extensions != null ? extensions : new HashMap<>();
     }
 
-    @Override
-    public Attributes getAttributes() {
-        return attributes;
-    }
-
     @Override
     public byte[] getData() {
         return this.data;
     }
 
+    @Override
+	public SpecVersion getSpecVersion() {
+		return this.attributes.getSpecVersion();
+	}
+
+	@Override
+	public String getId() {
+		return this.attributes.getId();
+	}
+
+	@Override
+	public String getType() {
+		return this.attributes.getType();
+	}
+
+	@Override
+	public URI getSource() {
+		return this.attributes.getSource();
+	}
+
+	@Override
+	public String getDataContentType() {
+		return this.attributes.getDataContentType();
+	}
+
+	@Override
+	public URI getDataSchema() {
+		return this.attributes.getDataSchema();
+	}
+
+	@Override
+	public String getSubject() {
+		return this.attributes.getSubject();
+	}
+
+	@Override
+	public ZonedDateTime getTime() {
+		return this.attributes.getTime();
+	}
+
     @Override
     public Map<String, Object> getExtensions() {
         return Collections.unmodifiableMap(extensions);
     }
 
-    @Override
+    Attributes getAttributes() {
+        return this.attributes;
+    }
+
     public CloudEvent toV03() {
```

Comment:
It would be nice if instead of `CloudEventImpl` with version converting methods there could instead be `V1CloudEvent` as the "impl" and a CloudEventBuilder could support conversion if a V1CloudEvent is passed in but V03 is requested via something like `builder.from(somev1CloudEvent).asV03()`.